src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -459,6 +459,46 @@ void GlobalAppSettings::_logSettingSet(const std::string_view& setting)
     }
 }
 
+void GlobalAppSettings::UpdateCommandID(const Model::Command& cmd, const winrt::hstring& newID)
+{
+    const auto oldID = cmd.ID();
+    _actionMap->UpdateCommandID(cmd, newID);
+    // newID might have been empty when this function was called, if so actionMap would have generated a new ID, use that
+    const auto updatedID = newID.empty() ? cmd.ID() : newID;

```

Comment:
```suggestion
void GlobalAppSettings::UpdateCommandID(const Model::Command& cmd, winrt::hstring newID)
{
    const auto oldID = cmd.ID();
    _actionMap->UpdateCommandID(cmd, newID);
    // newID might have been empty when this function was called. If so, ActionMap would have generated a new ID, use that
    newID = cmd.ID();
```

now you don't need to have a confusing local (`newId` vs `updatedId`)! it's OK to overwrite your own parameters after all